threads-web-ui/app/[username]/page.tsx

```diff
@@ -0,0 +1,87 @@
+import Image from "next/image";
+import React from "react";
+import { Thread, ThreadsIcons } from "react-threads";
+import { ThreadsAPI } from "threads-api";
+
+const UserProfilePage = async ({
+  params,
+}: {
+  params: { username: string };
+}) => {
+  const threadsAPI = new ThreadsAPI();
+  const username = params.username;
+
+  if (!username) {
+    return;
+  }
+
+  const userID = await threadsAPI.getUserIDfromUsername(username);
+  if (!userID) {
+    return <div>No user with the given username</div>;
+  }
+
+  const values: [string, string] = [username, userID];
+
+  const [userProfile, userThreads] = await Promise.all([
+    threadsAPI.getUserProfile(...values),
+    threadsAPI.getUserProfileThreads(...values),
+  ]);
```

Comment:
```suggestion
  const params: [string, string] = [username, userID];
  const [userProfile, userThreads] = await Promise.all([
    threadsAPI.getUserProfile(...params),
    threadsAPI.getUserProfileThreads(...params),
  ]);
```
`values` is way to general